optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/ScoreDirectorFactoryFactory.java

```diff
@@ -53,8 +53,11 @@
 
     private final ScoreDirectorFactoryConfig config;
 
-    public ScoreDirectorFactoryFactory(ScoreDirectorFactoryConfig config) {
+    private final KieBaseExtractor kieBaseExtractor;
+
+    public ScoreDirectorFactoryFactory(ScoreDirectorFactoryConfig config, KieBaseExtractor kieBaseExtractor) {
```

Comment:
Can we keep the original constructor as the default?
Users should only worry about the extractor when they need to, use default under the hood.